### PR TITLE
Code journal create entry

### PR DESCRIPTION
**Pull request by Bxvant**
Recording 2024-07-30 at 14 09 03
Recording 2024-07-30 at 14 29 02
Screenshot 2024-07-30 at 2 29 37 PM

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+.padding-20 {
+  padding: 10px;
+}
+
+/* below is the CSS for the body */
+
+.marginTopBottom10 {
+  margin-top: 10px;
+  margin-bottom: 10px;
+}
```
Comment (brandon-moy) [inline css/styles.css:17]: You could potentially shorten this to being `margin-y-10` or `my-10` to make the class shorter and still descriptive.

File: css/styles.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/* below is the CSS for the header */
+
+header {
+  background-color: #4c2676;
+  color: white;
+}
+
+.padding-20 {
+  padding: 10px;
+}
```
Comment (brandon-moy) [inline css/styles.css:10]: Why named `padding-20` when it's giving a padding of 10px? While yes it ends up being 20px on each side, but in total it would be 40px of padding. I would name this `padding-10` since it's 10px on each side.

File: ts/main.ts [hunk 3]
```diff
@@ -1 +1,44 @@
...
+/* code above is the event listener for the URL SRC */
+/* code below is the event listener for the submit button */
+
+const entryForm = document.querySelector('#entry-form') as HTMLFormElement;
+const $title = document.querySelector('#title') as HTMLInputElement;
+const $notes = document.querySelector('#notes') as HTMLTextAreaElement;
+
+if (!entryForm) throw new Error('$entryForm query has failed');
+if (!$title) throw new Error('$title query has failed');
+if (!$notes) throw new Error('$notes query has failed');
```
Comment (brandon-moy) [inline ts/main.ts:26]: While this does work, I would like you to get the value by accessing the FormControlCollection elements like we did in the `typescript-forms` exercise. The problem is that there may be a chance where the values did not update and we have old values. If we wait to access the form and values on submission, we will always get the most updated value from the form.

File: ts/main.ts [hunk 4]
```diff
@@ -1 +1,44 @@
...
+  event.preventDefault();
+  const title = $title.value;
+  const photoURL = $inputURL.value;
+  const notes = $notes.value;
+  const entry: Entry = {
+    title,
+    photoURL,
+    notes,
+    entryID: data.nextEntryId,
+  };
```
Comment (brandon-moy) [inline ts/main.ts:38]: This section will need to be adjusted based on comment above.

File: index.html [hunk 5]
```diff
@@ -17,11 +17,56 @@
...
+        <div class="column-full row">
+          <h1>New Entry</h1>
+        </div>
+        <form id="entry-form">
+          <div class="row">
+            <img
+              id="url-photo"
+              class="column-half"
+              src="/images/placeholder-image-square.jpg" />
+
```
Comment (brandon-moy) [inline index.html:38]: I believe your image is too large which is causing your notes section to be too long and adding scroll onto your desktop view. If you apply width directly to an image it will attempt to side based on the image, we want to size based on the `row`

**Review by brandon-moy — CHANGES_REQUESTED** [review]
Great start! Got some comments on things I would like cleaned up and also an extra on here:
- You inputs should have a purple border when focused
- Some CSS classes would be renamed for clarity and brevity
- Image sizing seems off and forcing your textarea to be too low on desktop view
- Rewrite getting form element values based on `typescript-forms` exercise

File: css/styles.css (hunk 2 above)
Comment (Bxvant) [reply css/styles.css:10]: i forgot to change the name, original padding value was 20px.

**Review by brandon-moy — APPROVED** [review]
Approved! ✅
